guac_core/src/channel_client/types.rs

```diff
@@ -6,20 +6,38 @@ use crypto::CryptoService;
 use std::ops::Add;
 use CRYPTO;
 
-#[derive(Copy, Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
-pub enum ChannelStatus {
-    Open,
-    Joined,
-    Challenge,
-    Closed,
+/// This is a state that is able to identity a channel uniquely.
+///
+/// A channel is registered with New state for a given address, which
+/// later identified with a channel id once it arrives in the contract.
```

Comment:
I don't know if it's useful to identify channels by their ID. Since there can only be one channel per counterparty, we should identify it by its counterparty. The ID is really more of a salt to prevent replay attacks.

Reply:
channels should be identified by counterparty everyone IMO it makes keeping references around hugely easier now that we have the 'now two channels' guarantee. The channel id is something to check, not something to index using.

Reply:
@jtremback @jkilpatr agreed. Before recent guac changes channels were identified and we thought it might be good idea to identify channels by ChannelId. Now it is no longer valid assumption so there is no reason to identify them by Address OR ChannelId when that exists. Identifying them by Address is enough for all cases and that also will make the code simplier.